src/opm/output/eclipse/AggregateNetworkData.cpp

```diff
@@ -0,0 +1,752 @@
+        }
+    }
+
+    // for nodes that are not well groups, set the node pressure to the fixed pressure potentially higher in the node tree
+    if (!node_wgroup) {
+        if (fixedPressureNode(sched, nodeName, lookup_step)) {
+            // node is a fixed pressure node
+            node_pres = units.from_si(M::pressure, network.node(nodeName).terminal_pressure().value());
+        }
+        else {
+            // find fixed pressure higher in the node tree
+            bool fp_flag = false;
+            auto node_name = nodeName;
+            auto upt_br = network.uptree_branch(node_name).value();
+            while (!fp_flag) {
+                if (fixedPressureNode(sched, upt_br.uptree_node(), lookup_step)) {
+                    node_pres = units.from_si(M::pressure, network.node(upt_br.uptree_node()).terminal_pressure().value());
+                    fp_flag = true;
+                } else {
+                    node_name = upt_br.uptree_node();
+                    if (network.uptree_branch(node_name).has_value()) {
+                        upt_br = network.uptree_branch(node_name).value();
+                    } else {
+                        std::stringstream str;
+                        str << "Node: " << nodeName << " has no uptree node with fixed pressure condition, uppermost node:  " << node_name;
+                        throw std::invalid_argument(str.str());
+                    }
+                }
+            }
+        }
+    }
+    return node_pres;
+}
+
+struct nodeProps
+{
+    double ndDeno;
+    double ndDenw;
+    double ndDeng;
+    double ndOpr;
+    double ndWpr;
+    double ndGpr;
+};
+
+nodeProps wellGroupRateDensity(const Opm::EclipseState&                  es,
+                               const Opm::Schedule&                      sched,
+                               const ::Opm::SummaryState&                smry,
+                               const std::string&                        groupName,
+                               const size_t                              lookup_step
+                              )
+{
+    auto get = [&smry](const std::string& well_name, const std::string& vector)
+    {
+        const auto key = vector + ':' + well_name;
+
+        return smry.has(key) ? smry.get(key) : 0.0;
+    };
+
+    const auto& stdDensityTable = es.getTableManager().getDensityTable();
+
+    double deno = 0.;
+    double denw = 0.;
+    double deng = 0.;
+    double opr  = 0.;
+    double wpr  = 0.;
+    double gpr  = 0.;
+    double t_opr = 0.;
+    double t_wpr = 0.;
+    double t_gpr = 0.;
+
+    const auto& grp = sched.getGroup(groupName, lookup_step);
+    // Calculate average flow rate and surface condition densities for wells in group
+    for (const auto& well_name : grp.wells()) {
+        const auto& well = sched.getWell(well_name, lookup_step);
+        if (well.isProducer()) {
+            const auto& pvtNum = well.pvt_table_number();
+            t_opr = get(well.name(),"WOPR");
```

Comment:
This should use: `smry.get_well_var(well.name(), "WOPR", 0.0);`

Reply:
OK - fine - done.